### PR TITLE
comments modifs

### PR DESCRIPTION
**Pull request by Fourchaux**
Continuation of https://github.com/c-cube/ocaml-containers/pull/310

### CODE REVIEW

File: src/core/CCListLabels.mli [hunk 1]
```diff
@@ -95,93 +96,99 @@ val fold_filter_map : f:('acc -> 'a -> 'acc * 'b option) -> init:'acc -> 'a list
...
 
 val count_true_false : f:('a -> bool) -> 'a list -> int * int
-(** @since 2.4 *)
+(** [count_true_false ~f l] returns a pair [(int1,int2)] where [int1] is the number of elements in [l]
+    that satisfy the predicate [~f], and [int2] the number of elements that do not satisfy [~f].
+    @since 2.4 *)
 
 val init : int -> f:(int -> 'a) -> 'a t
-(** [init len ~f] is [f 0; f 1; ...; f (len-1)].
+(** [init len ~f] is [~f 0; ~f 1; ⋯; ~f (len-1)].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:124]: Same in `[~f 0; ~f 1; ⋯; ~f (len-1)]`

File: src/core/CCListLabels.mli [hunk 2]
```diff
@@ -95,93 +96,99 @@ val fold_filter_map : f:('acc -> 'a -> 'acc * 'b option) -> init:'acc -> 'a list
...
 
 val flatten : 'a t t -> 'a t
-(** Safe flatten. Concatenate a list of lists. *)
-
+(** [flatten [l1]; [l2]; ⋯] concatenates a list of lists. 
+    Safe version of {!List.flatten}. *)
+    
 val product : f:('a -> 'b -> 'c) -> 'a t -> 'b t -> 'c t
-(** Cartesian product of the two lists, with the given combinator. *)
+(** [product ~f l1 l2] computes the cartesian product of the two lists, with the given combinator [~f]. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:184]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 3]
```diff
@@ -95,93 +96,99 @@ val fold_filter_map : f:('acc -> 'a -> 'acc * 'b option) -> init:'acc -> 'a list
...
 
 val fold_product : f:('c -> 'a -> 'b -> 'c) -> init:'c -> 'a t -> 'b t -> 'c
-(** Fold on the cartesian product. *)
-
+(** [fold_product ~f ~init l1 l2] applies the function [~f] with the accumulator [~init] on all
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:187]: Same in `[~f]` and `[~init]`

File: src/core/CCListLabels.mli [hunk 4]
```diff
@@ -197,34 +204,34 @@ val cartesian_product : 'a t t -> 'a t t
     @since 2.2 with labels *)
 
 val map_product_l : f:('a -> 'b list) -> 'a list -> 'b list list
-(** [map_product_l f l] maps each element of [l] to a list of
-    objects of type ['b] using [f].
-    We obtain [[l1;l2;...;ln]] where [length l=n] and [li : 'b list].
+(** [map_product_l ~f l] maps each element of [l] to a list of
+    objects of type ['b] using [~f].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:208]: Same here.

File: src/core/CCListLabels.mli [hunk 5]
```diff
@@ -197,34 +204,34 @@ val cartesian_product : 'a t t -> 'a t t
...
 val partition_map : f:('a -> [<`Left of 'b | `Right of 'c | `Drop]) ->
   'a list -> 'b list * 'c list
-(** [partition_map ~f l] maps [f] on [l] and gather results in lists:
-    - if [f x = `Left y], adds [y] to the first list.
-    - if [f x = `Right z], adds [z] to the second list.
-    - if [f x = `Drop], ignores [x].
+(** [partition_map ~f l] maps [~f] on [l] and gather results in lists:
+    - if [~f x = `Left y], adds [y] to the first list.
+    - if [~f x = `Right z], adds [z] to the second list.
+    - if [~f x = `Drop], ignores [x].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:223]: Same in `[~f]`.

File: src/core/CCListLabels.mli [hunk 6]
```diff
@@ -197,34 +204,34 @@ val cartesian_product : 'a t t -> 'a t t
...
     @since 0.11 *)
 
 val group_by : ?hash:('a -> int) -> ?eq:('a -> 'a -> bool) ->
   'a t -> 'a list t
-(** Group equal elements, regardless of their order of appearance.
-    precondition: for any [x] and [y], if [eq x y] then [hash x=hash y] must hold.
+(** [group_by ?hash ?eq l] groups equal elements, regardless of their order of appearance.
+    precondition: for any [x] and [y], if [~eq x y] then [~hash x = ~hash y] must hold.
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:229]: Same here.

File: src/core/CCListLabels.mli [hunk 7]
```diff
@@ -197,34 +204,34 @@ val cartesian_product : 'a t t -> 'a t t
...
     @since 2.3 *)
 
 val join : join_row:(('a -> 'b -> 'c option) [@keep_label]) -> 'a t -> 'b t -> 'c t
 (** [join ~join_row a b] combines every element of [a] with every
-    element of [b] using [join_row]. If [join_row] returns None, then
+    element of [b] using [~join_row]. If [~join_row] returns [None], then
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:234]: Same for `join_row`

File: src/core/CCListLabels.mli [hunk 8]
```diff
@@ -235,13 +242,13 @@ val join_by : ?eq:('key -> 'key -> bool) -> ?hash:('key -> int) ->
   'a t ->
   'b t ->
   'c t
-(** [join key1 key2 ~merge] is a binary operation
+(** [join_by ?eq ?hash key1 key2 ~merge la lb] is a binary operation
     that takes two sequences [a] and [b], projects their
     elements resp. with [key1] and [key2], and combine
     values [(x,y)] from [(a,b)] with the same [key]
-    using [merge]. If [merge] returns [None], the combination
+    using [~merge]. If [~merge] returns [None], the combination
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:249]: Same here

File: src/core/CCListLabels.mli [hunk 9]
```diff
@@ -235,13 +242,13 @@ val join_by : ?eq:('key -> 'key -> bool) -> ?hash:('key -> int) ->
...
     of values is discarded.
-    precondition: for any [x] and [y], if [eq x y] then [hash x=hash y] must hold.
+    precondition: for any [x] and [y], if [~eq x y] then [~hash x = ~hash y] must hold.
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:251]: Same here.

File: src/core/CCListLabels.mli [hunk 10]
```diff
@@ -250,13 +257,13 @@ val join_all_by : ?eq:('key -> 'key -> bool) -> ?hash:('key -> int) ->
...
   'c t
-(** [join_all_by key1 key2 ~merge] is a binary operation
+(** [join_all_by ?eq ?hash key1 key2 ~merge la lb] is a binary operation
     that takes two sequences [a] and [b], projects their
     elements resp. with [key1] and [key2], and, for each key [k]
     occurring in at least one of them:
     - compute the list [l1] of elements of [a] that map to [k]
     - compute the list [l2] of elements of [b] that map to [k]
-    - call [merge k l1 l2]. If [merge] returns [None], the combination
+    - call [~merge k l1 l2]. If [~merge] returns [None], the combination
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:266]: Same here.

File: src/core/CCListLabels.mli [hunk 11]
```diff
@@ -266,12 +273,12 @@ val group_join_by : ?eq:('a -> 'a -> bool) -> ?hash:('a -> int) ->
   'a t ->
   'b t ->
   ('a * 'b list) t
-(** [group_join_by key2] associates to every element [x] of
+(** [group_join_by ?eq ?hash key la lb] associates to every element [x] of
     the first sequence, all the elements [y] of the second
-    sequence such that [eq x (key y)]. Elements of the first
+    sequence such that [?eq x (key y)]. Elements of the first
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:278]: Same here.

File: src/core/CCListLabels.mli [hunk 12]
```diff
@@ -266,12 +273,12 @@ val group_join_by : ?eq:('a -> 'a -> bool) -> ?hash:('a -> int) ->
...
     sequences without corresponding values in the second one
     are mapped to [[]]
-    precondition: for any [x] and [y], if [eq x y] then [hash x=hash y] must hold.
+    precondition: for any [x] and [y], if [?eq x y] then [?hash x = ?hash y] must hold.
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:281]: Same here

File: src/core/CCListLabels.mli [hunk 13]
```diff
@@ -306,18 +313,18 @@ val sublists_of_len :
     @since 1.5 with labels *)
 
 val intersperse : x:'a -> 'a list -> 'a list
-(** Insert the first argument between every element of the list.
+(** [intersperse ~x l] inserts the element [~x] between adjacent elements of the list [l].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:316]: Same in `[~x]`

File: src/core/CCListLabels.mli [hunk 14]
```diff
@@ -347,15 +354,15 @@ val take_drop : int -> 'a t -> 'a t * 'a t
     [length l1 = min (length l) n]. *)
 
 val take_while : f:('a -> bool) -> 'a t -> 'a t
-(** [take_while ~f l] returns the longest prefix of [l] for which [f] is [true].
+(** [take_while ~f l] returns the longest prefix of [l] for which [~f] is [true].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:357]: Same here

File: src/core/CCListLabels.mli [hunk 15]
```diff
@@ -347,15 +354,15 @@ val take_drop : int -> 'a t -> 'a t * 'a t
...
     @since 0.13 *)
 
 val drop_while : f:('a -> bool) -> 'a t -> 'a t
-(** [drop_while ~f l] drops the longest prefix of [l] for which [f] is [true].
+(** [drop_while ~f l] drops the longest prefix of [l] for which [~f] is [true].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:361]: Same here

File: src/core/CCListLabels.mli [hunk 16]
```diff
@@ -364,55 +371,57 @@ val last : int -> 'a t -> 'a t
...
 val last_opt : 'a t -> 'a option
-(** Last element.
+(** [last_opt l] returns [Some x] (the last element of [l]) or [None] if the list [l] is empty. 
     @since 0.20 *)
 
 val find_pred : f:('a -> bool) -> 'a t -> 'a option
-(** [find_pred p l] finds the first element of [l] that satisfies [p],
-    or returns [None] if no element satisfies [p].
+(** [find_pred ~f l] finds the first element of [l] that satisfies [~f],
+    or returns [None] if no element satisfies [~f].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:389]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 17]
```diff
@@ -364,55 +371,57 @@ val last : int -> 'a t -> 'a t
...
 val find_mapi : f:(int -> 'a -> 'b option) -> 'a t -> 'b option
-(** Like {!find_map}, but also pass the index to the predicate function.
+(** [find_mapi ~f l] is like {!find_map}, but also pass the index to the predicate function.
     @since 0.11 *)
 
 val find_idx : f:('a -> bool) -> 'a t -> (int * 'a) option
-(** [find_idx p x] returns [Some (i,x)] where [x] is the [i]-th element of [l],
-    and [p x] holds. Otherwise returns [None]. *)
+(** [find_idx ~f x] returns [Some (i,x)] where [x] is the [i]-th element of [l],
+    and [~f x] holds. Otherwise returns [None]. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:414]: Same here.

File: src/core/CCListLabels.mli [hunk 18]
```diff
@@ -364,55 +371,57 @@ val last : int -> 'a t -> 'a t
...
 
 val remove : eq:(('a -> 'a -> bool) [@keep_label]) -> key:('a [@keep_label]) -> 'a t -> 'a t
     (* FIXME: the original CCList.mli uses ~x instead of ~key !! *)
-(** [remove ~key l] removes every instance of [key] from [l]. Tail-recursive.
+(** [remove ~eq ~key l] removes every instance of [~key] from [l]. Tail-recursive.
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:418]: Same for `key`

File: src/core/CCListLabels.mli [hunk 19]
```diff
@@ -364,55 +371,57 @@ val last : int -> 'a t -> 'a t
...
     @param eq equality function.
     @since 0.11 *)
 
 val filter_map : f:('a -> 'b option) -> 'a t -> 'b t
 (** [filter_map ~f l] is the sublist of [l] containing only elements for which
-    [f] returns [Some e].
+    [~f] returns [Some e].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:424]: Same here

File: src/core/CCListLabels.mli [hunk 20]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
     @since 2.2 with labels *)
 
 val sorted_merge : cmp:(('a -> 'a -> int) [@keep_label]) -> 'a list -> 'a list -> 'a list
-(** Merge elements from both sorted list. *)
-
+(** [sorted_merge ~cmp l1 l2] merges elements from both sorted list using
+    the given comparison function [~cmp]. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:452]: Same for `cmp`

File: src/core/CCListLabels.mli [hunk 21]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
+    
 val sort_uniq : cmp:(('a -> 'a -> int) [@keep_label]) -> 'a list -> 'a list
-(** Sort the list and remove duplicate elements. *)
-
+(** [sort_uniq ~cmp l] sorts the list [l] using the given comparison function [~cmp]
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:455]: Same for `[~cmp]`

File: src/core/CCListLabels.mli [hunk 22]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
+(** [uniq_succ ~eq l] removes duplicate elements that occur one next to the other.
     Examples:
     [uniq_succ [1;2;1] = [1;2;1]].
     [uniq_succ [1;1;2] = [1;2]].
     @since 0.10 *)
 
 val group_succ : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a list -> 'a list list
 (** [group_succ ~eq l] groups together consecutive elements that are equal
-    according to [eq].
+    according to [~eq].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:484]: Same here

File: src/core/CCListLabels.mli [hunk 23]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
     @since 0.11 *)
 
 (** {2 Indices} *)
 
 val mapi : f:(int -> 'a -> 'b) -> 'a t -> 'b t
-(** Like {!map}, but the function is applied to the index of
+(** [mapi ~f l] is like {!map}, but the function [~f] is applied to the index of
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:490]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 24]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
     the element as first argument (counting from 0), and the element
     itself as second argument. *)
 
 val iteri : f:(int -> 'a -> unit) -> 'a t -> unit
-(** Like {!iter}, but the function is applied to the index of
+(** [iteri ~f l] is like {!iter}, but the function [~f] is applied to the index of
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:495]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 25]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
     the element as first argument (counting from 0), and the element
     itself as second argument. *)
 
 val iteri2 : f:(int -> 'a -> 'b -> unit) -> 'a t -> 'b t -> unit
-(** Iter on two lists.
+(** [iteri2 ~f l1 l2] applies [~f] to the two lists [l1] and [l2] simultaneously.
+    The integer passed to [~f] indicates the index of element.
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:501]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 26]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
     @raise Invalid_argument when lists do not have the same length.
     @since 2.0, but only
     @since 2.2 with labels *)
 
 val foldi : f:('b -> int -> 'a -> 'b) -> init:'b -> 'a t -> 'b
-(** Like [fold] but it also passes in the index of each element to the folded function. Tail-recursive. *)
-
+(** [foldi ~f ~init l] is like [fold] but it also passes in the index of each element, from [0] to [length l - 1]
+    as additional argument to the folded function [~f]. Tail-recursive. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:508]: Same in `[~f]`

File: src/core/CCListLabels.mli [hunk 27]
```diff
@@ -439,99 +448,107 @@ val all_ok : ('a, 'err) result t -> ('a t, 'err) result
...
+    
 val foldi2 : f:('c -> int -> 'a -> 'b -> 'c) -> init:'c -> 'a t -> 'b t -> 'c
-(** Fold on two lists, with index.
+(** [foldi2 ~f ~init l1 l2] folds on the two lists [l1] and [l2],
+    with index of each element passed to the function [~f].
+    Computes [~f(… (~f ~init i_0 l1_0 l2_0) …) i_n l1_n l2_n] .
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:513]: Same for `f`

File: src/core/CCListLabels.mli [hunk 28]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
 val remove_one : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a -> 'a t -> 'a t
-(** [remove_one x set] removes one occurrence of [x] from [set]. Linear time.
+(** [remove_one ~eq x set] removes one occurrence of [x] from [set]. Linear time.
     @since 0.11 *)
 
 val mem : ?eq:(('a -> 'a -> bool) [@keep_label]) -> 'a -> 'a t -> bool
-(** Membership to the list. Linear time. *)
-
+(** [mem ?eq x l] is [true] iff [x] is equal to an element of [l].
+    A comparator function [~eq] can be provided. Linear time. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:570]: Same in `[~eq]`

File: src/core/CCListLabels.mli [hunk 29]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
+    
 val subset : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a t -> 'a t -> bool
-(** Test for inclusion. *)
-
+(** [subset ~eq l1 l2] tests if all elements of the list [l1] are contained
+    in the list [l2] by applying [~eq]. *)
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:574]: Same in `[~eq]`

File: src/core/CCListLabels.mli [hunk 30]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
+    
 val uniq : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a t -> 'a t
-(** Remove duplicates w.r.t the equality predicate.
+(** [uniq ~eq l] removes duplicates in [l] w.r.t the equality predicate [~eq].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:577]: Same in `[~eq]`

File: src/core/CCListLabels.mli [hunk 31]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
     Complexity is quadratic in the length of the list, but the order
     of elements is preserved. If you wish for a faster de-duplication
     but do not care about the order, use {!sort_uniq}. *)
 
 val union : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a t -> 'a t -> 'a t
-(** List union. Complexity is product of length of inputs. *)
+(** [union ~eq l1 l2] is the union of the lists [l1] and [l2] w.r.t. the equality predicate [~eq].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:583]: Same in `[~eq]`

File: src/core/CCListLabels.mli [hunk 32]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
+    Complexity is product of length of inputs. *)
 
 val inter : eq:(('a -> 'a -> bool) [@keep_label]) -> 'a t -> 'a t -> 'a t
-(** List intersection. Complexity is product of length of inputs. *)
-
+(** [inter ~eq l1 l2] is the intersection of the lists [l1] and [l2] w.r.t. the equality predicate [~eq].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:587]: Same in `[~eq]`

File: src/core/CCListLabels.mli [hunk 33]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
 (** {2 Other Constructors} *)
 
 val range_by : step:(int [@keep_label]) -> int -> int -> int t
 (** [range_by ~step i j] iterates on integers from [i] to [j] included,
-    where the difference between successive elements is [step].
-    Use a negative [step] for a decreasing list.
-    @raise Invalid_argument if [step=0].
+    where the difference between successive elements is [~step].
+    Use a negative [~step] for a decreasing list.
+    @raise Invalid_argument if [~step=0].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:596]: Same in `step`

File: src/core/CCListLabels.mli [hunk 34]
```diff
@@ -541,112 +558,127 @@ val remove_at_idx : int -> 'a t -> 'a t
...
   val mem : ?eq:(('a->'a->bool) [@keep_label]) -> 'a -> ('a,_) t -> bool
-  (** [mem x l] returns [true] iff [x] is a key in [l].
+  (** [mem ?eq k alist] returns [true] iff [k] is a key in [alist].
       @since 0.16 *)
-
+      
   val update :
     eq:(('a->'a->bool) [@keep_label]) -> f:(('b option -> 'b option) [@keep_label]) -> 'a -> ('a,'b) t -> ('a,'b) t
-  (** [update k ~f l] updates [l] on the key [k], by calling [f (get l k)]
+  (** [update ~eq ~f k alist] updates [alist] on the key [k], by calling [~f (get alist k)]
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:645]: Same in `[~f (get alist k)]`

File: src/core/CCListLabels.mli [hunk 35]
```diff
@@ -695,9 +731,8 @@ module Traverse(M : MONAD) : sig
   val map_m : f:('a -> 'b M.t) -> 'a t -> 'b t M.t
 
   val map_m_par : f:('a -> 'b M.t) -> 'a t -> 'b t M.t
-  (** Like {!map_m} but [map_m_par f (x::l)] evaluates [f x] and
-      [f l] "in parallel" before combining their result (for instance
-      in Lwt). *)
+  (** [map_m_par ~f (x :: l)] is like {!map_m} but [~f x] and [~f l] are evaluated
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:734]: Same for `f`

File: src/core/CCListLabels.mli [hunk 36]
```diff
@@ -707,51 +742,51 @@ val random_non_empty : 'a random_gen -> 'a t random_gen
...
     @raise Not_found if the list is empty. *)
 
 val random_sequence : 'a random_gen t -> 'a t random_gen
 
 val to_string : ?start:string -> ?stop:string -> ?sep:string ->
   ('a -> string) -> 'a t -> string
-(** [to_string ~start ~stop ~sep item_to_string l] print [l] to a string using
-    [sep] as a separator between elements of [l].
+(** [to_string ?start ?stop ?sep item_to_string l] print [l] to a string using
+    [?sep] as a separator between elements of [l].
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:753]: Same in `[?sep]`

File: src/core/CCList.mli [hunk 37]
```diff
@@ -745,8 +745,8 @@ val random_sequence : 'a random_gen t -> 'a t random_gen
 
 val to_string : ?start:string -> ?stop:string -> ?sep:string ->
   ('a -> string) -> 'a t -> string
-(** [to_string ~start ~stop ~sep item_to_string l] prints [l] to a string using
-    [sep] as a separator between elements of [l].
+(** [to_string ?start ?stop ?sep item_to_string l] prints [l] to a string using
+    [?sep] as a separator between elements of [l].
```
[the remark this answers is not in the document]
Comment (c-cube) [reply src/core/CCList.mli:749]: the arguments are optional, so you can pass them with `?start` if `start:string option`

**Comment by Fourchaux** [comment]
Thanks @FardaleM for this meticulous review.

All these [~foo] are used in (and come from) CCArrayLabels.mli.

So I'll apply your suggestions to CCListLabels.mli and also modify CCArrayLabels.mli.

File: src/core/CCListLabels.mli [hunk 38]
```diff
@@ -580,73 +601,84 @@ val range : int -> int -> int t
...
     @since 1.5, but only
     @since 2.0 with labels *)
 
 val mem_assoc : ?eq:(('a -> 'a -> bool) [@keep_label]) -> 'a -> ('a * _) t -> bool
-(** Like [Assoc.mem].
+(** [mem_assoc ?eq k alist] returns [true] iff [k] is a key in [alist].
+    Like [Assoc.mem].
     @since 2.0 *)
-
+
```
Comment (FardaleM) [inline src/core/CCListLabels.mli:678]: Some spaces have sneaked in.

File: src/core/CCList.mli (hunk 37 above)
[the remark this answers is not in the document]
Comment (FardaleM) [reply src/core/CCList.mli:749]: Yes, I means in `[?sep]`

**Comment by Fourchaux** [comment]
Corrected.

**Comment by FardaleM** [comment]
Perfect!

**Pull request merged**